src/server/worker.cc

```diff
@@ -344,6 +344,31 @@ redis::Connection *Worker::removeConnection(int fd) {
   return conn;
 }
 
+// MigrateConnection moves the connection to another worker
+// when reducing the number of workers.
+//
+// To make it simple, we would close the connection if it's
+// blocked on a key or stream.
+void Worker::MigrateConnection(Worker *target, redis::Connection *conn) {
+  if (!target || !conn) return;
+  if (conn->current_cmd != nullptr && conn->current_cmd->IsBlocking()) {
+    // don't need to close the connection since destroy worker thread will close it
+    return;
+  }
+
```

Comment:
Could any blocking command be received after the previous check and before the actual connection migrating?

Reply:
It's not since Config::Set is an exclusive command, so it won't receive any other commands before this process was finished.